server/lib/datadog-collector.js

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var logger = require('mozlog')('server.datadog');
+var StatsD = require('node-statsd');
+// TODO: set configuration options for dev / stage / prod Datadog
+var host = 'localhost';
+var port = 8125;
+
+var client = new StatsD(host, port);
+
+if (client.socket) {
+  client.socket.on('error', function (error) {
+    return logger.error('Error in stats socket: ', error);
+  });
+} else {
+  logger.error('Datadog failed to connect to ' + host + ':' + port);
+}
+
+
+function DataDogCollector() {
+  // nothing to do here.
+}
+
+DataDogCollector.prototype = {
+  send: function (body) {
+    if (body && body.events && body.events.length > 0) {
+      logger.verbose(body.events);
+      // see more about tags here: http://docs.datadoghq.com/guides/metrics/
+      var tags = [
+        'campaign:' + body.campaign,
+        'context:' + body.context,
+        'entrypoint:' + body.entrypoint,
+        'migration:' + body.migration,
+        'lang:' + body.lang,
+        'marketingClicked:' + body.marketingClicked,
+        'marketingLink:' + body.marketingLink,
+        'marketingType:' + body.marketingType,
+        'screen-devicePixelRatio:' + body.devicePixelRatio,
+        'screen-clientWidth:' + body.clientWidth,
+        'screen-clientHeight:' + body.clientHeight,
+        'screen-width:' + body.width,
+        'screen-height:' + body.height,
+        'service:' + body.service,
+        // TODO: agent needs to be tested or use a UA parser to properly create tags
+        'agent:' + body.agent
+      ];
+      var sampleRate = 1;
```

Comment:
Oh

> rate: a sample rate, a float between 0 and 1. Will only send data this percentage of the time. The statsd server will take the sample rate into account for counters.

I thought it was about telling it how much we collect, but yea that means this will be configurable